src/test/lib/updating-element_test.ts

```diff
@@ -1802,6 +1802,134 @@ suite('UpdatingElement', () => {
         assert.equal(sub.getAttribute('foo'), '5');
       });
 
+    test('can provide a default property declaration', async () => {
```

Comment:
Just pointing out, now that `getDeclarationForProperty` is exposed, this use case _could_ be covered by overriding that method, albeit less efficiently (O.assign on every get).  But might want to double-check the code is safe to someone doing that, i.e. the map is never accessed directly but always through that method?

Reply:
Added doc note that it should not be implemented.